### ISSUE TITLE
MediaConvert.createJob does not use queue configured in jobTemplate if CreateJobRequest.JobTemplate is specified

### ISSUE THREAD

**Issue by CCattano** [aws/aws-sdk-js-v3#5248]
### Checkboxes for prior research

- [X] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest)
- [X] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js).
- [X] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-js-v3/issues) and didn't find any solution.

### Describe the bug

When creating a job via `MediaConvert.createJob`.

If a job template to use is specified via `MediaConvert.Types.CreateJobRequest.JobTemplate`, and the template is configured to use a specific queue.

The job created by the SDK does not get submitted to the queue defined by the job template.

### SDK version number

@aws-sdk@^2.1221.0 -> Currently the ^ in our package.json has npm resolving to v2.1353 

### Which JavaScript Runtime is this issue in?

Node.js

### Details of the browser/Node.js/ReactNative version

node v16.19.1

### Reproduction Steps

1. Create Media Convert Job Queue in AWS Online Console
2. Create Media Convert Job Template in AWS Online Console
    a. Specify the job template should send jobs to the queue made in step 1
3. Use something akin to code in the "Observed Behaviour" description to programmatically generate the job
    a. Making sure to specify a `JobTemplate` in `CreateJobRequest` but not a `Queue`
5. Go to Media Convert Jobs list in AWS Online Console
6. See the job created by the SDK does not use the queue configured in the template

### Observed Behavior

I am integrating with the AWS MediaConvert SDK as follows

```typescript
/**
 * Executes a defined jobTemplate
 * @param jobTemplateName
 * @param inputFileLocation
 * @param outputFileLocation
 */
public async executeJob(
  jobTemplateName: string,
  inputFileLocation: string,
  outputFileLocation: string
): Promise<CreateJobResponse> {
  const params: CreateJobRequest = {
    JobTemplate: jobTemplateName,
    Role: 'arn:aws:iam::12345:role/Role_With_RelevantPerms',
    Settings: {
      OutputGroups: [{
        OutputGroupSettings: {
          Type: 'HLS_GROUP_SETTINGS',
          HlsGroupSettings: {
            Destination: outputFileLocation
          }
        }
      }],
      Inputs: [{
        FileInput: inputFileLocation
      }]
    }
  };

  try {
    const data: CreateJobResponse =
      await this._awsMediaConvertClient.createJob(params).promise();

    return data;
  } catch (err) {
    throw new CreateMediaConvertJobException(); // this is an in-house exception
  }
}
```

When I specify a job template to create my job from, that template is configured in AWS Online Console to use a specific queue.

Even though I've specified the job template to use here in my SDK integration the queue in that job template is not being honored when the SDK creates the job for me.

I have six job templates. Three point to one queue while the other three point to a second queue.

It seems that because I have not provided a value in the `CreateJobRequest.Queue` property the SDK is having my jobs be put on the default queue.

### Expected Behavior

I would expect that every aspect of my job template be utilized if specified, including the queue to use.

I'm not sure if this a bug or maybe a technical limitation of the AWS API that the SDK is tapping into on my behalf.

### Possible Solution

When a job template is specified and configured to utilize a specific job queue the job created by the SDK should be submitted to the queue defined in the template.

**Comment by RanVaknin** [comment]
Hi @CCattano,

Before we start diving into reproducing this I see you are using a very old version of the SDK. 
Can you please upgrade your SDK version and try again? I'd say for the best result you might want to delete `node_modules` and `package-lock.json` and then run `npm install` to make sure all the transitive dependencies also get updated correctly. 

Thanks,
Ran~

**Comment by CCattano** [comment]
Hey @RanVaknin, thanks for getting back to me so quickly. Sorry I left you hanging here. By time the weekend had ended I completely forgot I submitted this and failed to circle back promptly, my bad.

Unfortunately upgrading our infrastructure to v3 is out of scope for my current work. We are also integrated with a lot of SDK services besides MediaConvert so it would not be a quick update on my end.

Upgrading all our SDK integrations would touch a large swath of our product and require a large allocation of resources for comprehensive regression testing and we're not at a point where the team is prioritizing tech debt.

The warning the library prints during compilation states, "(node:32560) NOTE: We are formalizing our plans to enter AWS SDK for JavaScript (v2) into maintenance mode in 2023."

I guess idk exactly what "maintenance" entails, but if it means outstanding issues aren't being investigated b/c the library version is deprecated then no worries.

For what it's worth this does not block us from using the library, I just had to manually specify the queue like so.

```typescript

const jobQueueToUse: string = shouldUseQueueA
  ? 'arn:aws:mediaconvert:us-east-1:12345:queues/queue-a'
  : 'arn:aws:mediaconvert:us-east-1:12345:queues/queue-b'

const params: CreateJobRequest = {
  JobTemplate: jobTemplateName,
  Queue: jobQueueToUse,
  Role: 'arn:aws:iam::12345:role/Role_With_RelevantPerms',
  Settings: {
    OutputGroups: [{
      OutputGroupSettings: {
        Type: 'HLS_GROUP_SETTINGS',
        HlsGroupSettings: {
          Destination: outputFileLocation
        }
      }
    }],
    Inputs: [{
      FileInput: inputFileLocation
    }]
  }
};
```

I suppose my submitted issue is less a bug report and more meant to bring attention to this behaviour in case someone, "in the know" was able to identify it is actually unexpected behaviour and not how the library was meant to function.

It just felt counterintuitive and like a bug to myself from my limited perspective given the job template contains this info so I didn't expect I should have to reiterate it.

The reproduction steps laid out in the OP are fairly high-level steps/concepts and I don't think they would be version-specific. They would probably apply to v3 as well if you want to verify this still occurs in v3. I suspect it likely does but am unable to test that myself.

Either way, if this is an issue that can't be investigated further at this time b/c of the version it's occurring in I understand.

Once we prioritize upgrading all our AWS SDK integrations to v3 I can circle back to see if this is still happening if it's not investigated beyond this point for anyone who may be encountering it in v2 or v3.

**Comment by RanVaknin** [comment]
Hi @CCattano ,

Thanks for the follow up. 
I was under the impression you are using SDK js V3 since you mentioned "@aws-sdk" for version, and opened this in the v3 repo, however based on the error message you are seeing you have some `aws-sdk` (v2) imports which would explain why the version number is so low - lol.

>I guess idk exactly what "maintenance" entails, but if it means outstanding issues aren't being investigated b/c the library version is deprecated then no worries.

Maintenance means the SDK will only be getting critical fixes and security updates, but not new features. Its a step before deprecation. You can read more about it [here](https://docs.aws.amazon.com/sdkref/latest/guide/maint-policy.html), but we are not even there yet. JS SDK v2 is still GA, but I highly recommend moving to v3 when possible.

If I understand the expected behavior here is that the job should be submitted to the queue as specified in the job template. Instead, it's being sent to the default queue? Unless the SDK is not serializing any information specified in your inputs, this sounds like a service related bug, and not an SDK bug.

The easiest way to know would be to attach a request handler and log the outbound serialized raw http request:

```js
const request =  mediaconvert.createJob(createJobParams)

request.on('httpHeaders', function(statusCode, headers) {
    console.log('Serialized Request:', headers);
  });

console.log(request.send())
```

This should log something like this:

```
params: {
      JobTemplate: 'foo-bar-template',
      Role: 'arn:aws:iam::REDACTED:role/MediaConvertRole',
      Settings: [Object],
      ClientRequestToken: 'REDACTED'
    },
    httpRequest: HttpRequest {
      method: 'POST',
      path: '/2017-08-29/jobs',
      headers: [Object],
      body: '{"jobTemplate":"foo-bar-template","role":"arn:aws:iam::REDACTED:role/MediaConvertRole","settings":{"outputGroups":[{"outputGroupSettings":{"type":"HLS_GROUP_SETTINGS","hlsGroupSettings":{"destination":"s3://my-output-location/"}}}],"inputs":[{"fileInput":"s3://my-input-file-location/"}]},"clientRequestToken":"REDACTED"}',
      endpoint: [Object],
      region: 'us-east-1',
      _userAgent: 'aws-sdk-nodejs/2.1467.0 darwin/v19.9.0',
      stream: [ClientRequest]
    },
```
To me this seems like the SDK is serializing the request correctly, but it would be nice if you can verify on your end. 

You can also isolate the issue by using the CLI with the same input and observing the behavior. Let me know.

Thanks,
Ran~
